aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java

```diff
@@ -41,23 +41,18 @@ void setReadHandler(final ReadHandler readHandler) {
             final CallbackContext callbackContext,
             final Logger logger) {
 
-        final List<ResourceModel> models = new ArrayList<>();
+        DescribePatchBaselinesRequest describePatchBaselinesRequest = DescribePatchBaselinesRequest.builder().nextToken(request.getNextToken()).maxResults(MAX_RESULTS).build();
 
-        DescribePatchBaselinesResponse describePatchBaselinesResponse = describePatchBaselinesResponse(proxy);
+        final DescribePatchBaselinesResponse describePatchBaselinesResponse = proxy.injectCredentialsAndInvokeV2(describePatchBaselinesRequest, ssmClient::describePatchBaselines);
 
-        if (describePatchBaselinesResponse.hasBaselineIdentities()) {
-            models.addAll(getResourceModelFromResponse(describePatchBaselinesResponse, request, proxy, logger));
-        }
+        final List<ResourceModel> models = describePatchBaselinesResponse
+                .baselineIdentities()
+                .stream().map(baseline -> ResourceModel.builder().id(baseline.baselineId()).name(baseline.baselineName()).operatingSystem(baseline.operatingSystemAsString()).description(baseline.baselineDescription()).build()).collect(Collectors.toList());
```

Comment:
what would happen if one of the properties don't exist? the returned properties only baselinId, OS are guaranteed to exist.

Reply:
1.) Here the Listhandler will return the baseline details,  The baseline details we are getting from the describePatchBaseline . The describePatchBaseline is retuning  the basics details of baselineInformation, Like baselined, baselineName, operating system details, BaselineDescription. 

2. From the list of baselineEntities, we are getting these four properties and we are returning the these properties as an ResourceModel. These four properties required properties I think so. If the properties does not have, it will return empty for those baseline entities.

Reply:
does the unit test cover this case?  I have second thoughts of adding empty string to non-exist properties from describePatchBaseline, you can decide which one to keep.

Reply:
yes this was good with unit testcase. 
Instead of ignoring the properties, we can add the property and will send with empty string..
This one we can keep and already i have done the same

Reply:
i mean add a specific unit test with no property of say Description

Reply:
I don't think the unit test actually test this case, and i have doubt in the correctness of the unit test, i wrote it doesn't mean it's the right unit test, I don't understand the unit test logic now, and it's not asserting the expected results. Can you work on the unit test to make sure it's testing what it should test

Reply:
I have added the unit test and validated with invalid baseline id, the test case will returns the values as below and test case was successful

As i told previous, the if needed property was missed it will not return as empty and it will return as null as below.

{
BaslineId : "PD-acdescrfed01",
"baelineName" " "Window-Patch-baseline"
Description : null,
OperatingSysetm: "Window"

}


Output : 
ProgressEvent(status=SUCCESS, errorCode=null, message=null, callbackContext=null, callbackDelaySeconds=0, resourceModel=null, resourceModels=[ResourceModel(id=pb-12345678901234567, defaultBaseline=null, operatingSystem=null, description=null, approvalRules=null, sources=null, name=null, rejectedPatches=null, approvedPatches=null, rejectedPatchesAction=null, patchGroups=null, approvedPatchesComplianceLevel=null, approvedPatchesEnableNonSecurity=null, globalFilters=null, tags=null)], nextToken=null)

TestCase output :
[INFO] Running software.amazon.ssm.patchbaseline.ListHandlerTest
ProgressEvent(status=SUCCESS, errorCode=null, message=null, callbackContext=null, callbackDelaySeconds=0, resourceModel=null, resourceModels=[ResourceModel(id=pb-12345678901234567, defaultBaseline=null, operatingSystem=null, description=null, approvalRules=null, sources=null, name=null, rejectedPatches=null, approvedPatches=null, rejectedPatchesAction=null, patchGroups=null, approvedPatchesComplianceLevel=null, approvedPatchesEnableNonSecurity=null, globalFilters=null, tags=null)], nextToken=null)
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.046 s - in software.amazon.ssm.patchbaseline.ListHandlerTest
[INFO]

Reply:
In the list handler, we are using the describePatchBasline to get the baselineId’s with all entities. what the patch baseline process is providing and  We are getting the same baseline id details,

To test the the exception case , I tired to pass the invalid baseline id to describepathbasline model and send that model to list-handler invoke process, which will get the describe patchbasleline details, and we are not getting the exception.

Sample:

final ResourceModel model = ResourceModel.builder().id("pb-12345678901234567").build();

        final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel>builder()

                .desiredResourceState(model)

                .clientRequestToken(CLIENT_REQUEST_TOKEN)

                .build();



        final ProgressEvent<ResourceModel, CallbackContext> response =

                listHandler.handleRequest(proxy, request, null, logger);

If the baseline id does not exits in the patch baseline process, it will return baseline id details and remaining fields it will return as null instead of empty string. Since all the properties are strings so the process itself is returning null value.
For this response we don’t have any control in our hand, but we can do the translation after response got.

But in the original code also , we are not using any translator since u have ask me to send the response what we are getting the response from describepatchbaseline and I am sending the same.

In the above code  snippet it will return the response as below,

ProgressEvent(status=SUCCESS, errorCode=null, message=null, callbackContext=null, callbackDelaySeconds=0, resourceModel=null, resourceModels=[ResourceModel(id=pb-12345678901234567, defaultBaseline=null, operatingSystem=null, description=null, approvalRules=null, sources=null, name=null, rejectedPatches=null, approvedPatches=null, rejectedPatchesAction=null, patchGroups=null, approvedPatchesComplianceLevel=null, approvedPatchesEnableNonSecurity=null, globalFilters=null, tags=null)], nextToken=null)

To test the above case, I have created the resource model as below,

ResourceModel expectedModel = ResourceModel.builder().id("pb-12345678901234567").operatingSystem(null).description(null).name(null).build();

I am assuming if the baseline id is invalid it should return the response fields as I mentioned above and it won’t return exception.

Testcase sample:

 if(response.getResourceModels().size() > 0){

            assertThat(response.getResourceModels().get(0)).isEqualTo(expectedModel);

        }
In the above snippet, After list handler invoking process, it will return the baseline details as list of values, so I have verifying if the list has elements ar not. And I am getting first element from the list and validating the response with my expected model


        assertThat(response.getResourceModels()).contains(expectedModel);